README.md
# Abandoned-MMORPG-Project
*An MMORPG Unity project initialized in 2011, developed for 3 years and eventually abandoned...*

**Alright, some context!**

This was the classic "my first game will be a huge mmo success" attempt, condemned ofcourse to the same end as any other. I started this project in the first year of high school, when i started learning programming along with 3d graphics after working with flash and actionscript 3 for some time. Since it was my first big programming project it started as some expiremental scenes and code, deleting and re-creating everything multiple times to reach the current state. During the 3 years of its development a few very talented people added their artistic touch too. Although most of you will consider this project a messy piece of trash, there is something "unique" about it. Every map it contains was concieved in a dream. I didn't have the easiest sleep back then and used to dream different landscapes and stories very vividly sometimes. Some of them are in here. The project was abandoned when i entered the university and some maps didn't make it. The repository contains all the artwork i possess too in hope someone will see what i saw when i was making it.

***DISCLAIMER***

This is a project developed by an underage game-design-obsessed kid. Although i consider myself organizing dont expect to find any professional work in there. When i left the project:

*• Unity version was 4.5*

*• There were no errors (I opened it before upload it and there are only deprecation errors and warnings)*

*• Everything was organized in folders, without any missing assets (There are some objects in the scenes with 'missing' materials but im sure they are there. Even if they are not they are easily replacable*

*• There were no nested prefabs, and the way i made the prefabs is probably not very efficient, totally simple to use though*

*• The scene objects were not organized by group or by name*

*• There are used Asset Store plug-ins that are all obviously deprecated or not up-to-date*

*• The code is not commented but configurable by the script names and objects they are on*

*• I have only left the 'Assets' and 'Packages' folders so when you open it for first time Unity will rebuild the rest*

*• The server-side code is lost. But it doesn't matter since the whole code is ancient anyway*

*• Code-wise the game has very extended abilities considering the circumstances, like inventory (drag'n'drop) system, quest system, stats mathematical model (simple, im shit at math), an awesome rotating Pirates of the Caribbean 3 style map, enemy ai (again simple) and guess what, working networking! You could create an account back then and sync with other players and monsters. Complete with SQL, PHP, JS, SmartfoxServer X2 (i think its deprecated) and all!*

*Below you can see a sample of the map. Incredible work of [Kenji Iwata](https://iwatakenji.wordpress.com/):*
![Map](https://github.com/Mistral-Designs/The_Abandoned_MMORPG_Project/blob/master/map.jpg?raw=true)

There is much more content like icons, story, quests etc...

If you want my opinion there is no sense in "continuing" this project. If i wanted to work on it i would only use it as an "artistic guide". Even the code is too old to copy. The maps may not compare to the current ones Unity can make, hell probably didn't compare even back then, but they tell something. I would also be a bit careful with the third-party licenses (yo-ho)... If you catch the world i had in my mind back then, even now you can have an amazing result. Im sorry this project didn't make it. It was obviously doomed from the start but i put many many hours and love into it. I have currently left the gaming industry for good but it would be a dream to see it make flesh one day even in some other's hands with the same obsession i had. In that case please contact me, i would love to help at least concept-wise.


history.md
Contribution: 2020-12-10 15:54

Contribution: 2020-12-10 15:55

Contribution: 2020-12-10 15:56

Contribution: 2020-12-10 15:57

Contribution: 2020-12-10 15:58

Contribution: 2020-12-10 15:59

Contribution: 2020-12-11 15:54

Contribution: 2020-12-11 15:55

Contribution: 2020-12-14 15:54

Contribution: 2020-12-15 15:54

Contribution: 2020-12-16 15:54

Contribution: 2020-12-16 15:55

Contribution: 2020-12-16 15:56

Contribution: 2020-12-16 15:57

Contribution: 2020-12-16 15:58

Contribution: 2020-12-16 15:59

Contribution: 2020-12-16 16:00

Contribution: 2020-12-21 15:54

Contribution: 2020-12-21 15:55

Contribution: 2020-12-21 15:56

Contribution: 2020-12-21 15:57

Contribution: 2020-12-22 15:54

Contribution: 2020-12-22 15:55

Contribution: 2020-12-22 15:56

Contribution: 2020-12-22 15:57

Contribution: 2020-12-22 15:58

Contribution: 2020-12-22 15:59

Contribution: 2020-12-23 15:54

Contribution: 2020-12-23 15:55

Contribution: 2020-12-23 15:56

Contribution: 2020-12-23 15:57

Contribution: 2020-12-24 15:54

Contribution: 2020-12-24 15:55

Contribution: 2020-12-24 15:56

Contribution: 2020-12-24 15:57

Contribution: 2020-12-24 15:58

Contribution: 2020-12-28 15:54

Contribution: 2020-12-28 15:55

Contribution: 2020-12-28 15:56

Contribution: 2020-12-28 15:57

Contribution: 2020-12-28 15:58

Contribution: 2020-12-28 15:59

Contribution: 2020-12-29 15:54

Contribution: 2020-12-29 15:55

Contribution: 2020-12-29 15:56

Contribution: 2020-12-29 15:57

Contribution: 2020-12-30 15:54

Contribution: 2020-12-30 15:55

Contribution: 2020-12-30 15:56

Contribution: 2021-01-01 15:54

Contribution: 2021-01-01 15:55

Contribution: 2021-01-04 15:54

Contribution: 2021-01-04 15:55

Contribution: 2021-01-04 15:56

Contribution: 2021-01-05 15:54

Contribution: 2021-01-05 15:55

Contribution: 2021-01-06 15:54

Contribution: 2021-01-06 15:55

Contribution: 2021-01-06 15:56

Contribution: 2021-01-06 15:57

Contribution: 2021-01-07 15:54

Contribution: 2021-01-11 15:54

Contribution: 2021-01-11 15:55

Contribution: 2021-01-11 15:56

Contribution: 2021-01-11 15:57

Contribution: 2021-01-13 15:54

Contribution: 2021-01-13 15:55

Contribution: 2021-01-13 15:56

Contribution: 2021-01-13 15:57

Contribution: 2021-01-13 15:58

Contribution: 2021-01-14 15:54

Contribution: 2021-01-14 15:55

Contribution: 2021-01-14 15:56

Contribution: 2021-01-14 15:57

Contribution: 2021-01-14 15:58

Contribution: 2021-01-14 15:59

Contribution: 2021-01-15 15:54

Contribution: 2021-01-15 15:55

Contribution: 2021-01-18 15:54

Contribution: 2021-01-18 15:55

Contribution: 2021-01-18 15:56

Contribution: 2021-01-18 15:57

Contribution: 2021-01-18 15:58

Contribution: 2021-01-18 15:59

Contribution: 2021-01-18 16:00

Contribution: 2021-01-19 15:54

Contribution: 2021-01-19 15:55

Contribution: 2021-01-19 15:56

Contribution: 2021-01-19 15:57

Contribution: 2021-01-19 15:58

Contribution: 2021-01-19 15:59

Contribution: 2021-01-22 15:54

Contribution: 2021-01-22 15:55

Contribution: 2021-01-22 15:56

Contribution: 2021-01-25 15:54

Contribution: 2021-01-26 15:54

Contribution: 2021-01-26 15:55

Contribution: 2021-01-28 15:54

Contribution: 2021-01-28 15:55

Contribution: 2021-01-28 15:56

Contribution: 2021-01-28 15:57

Contribution: 2021-01-28 15:58

Contribution: 2021-01-28 15:59

Contribution: 2021-01-28 16:00

Contribution: 2021-01-29 15:54

Contribution: 2021-01-29 15:55

Contribution: 2021-01-29 15:56

Contribution: 2021-01-29 15:57

Contribution: 2021-01-29 15:58

Contribution: 2021-01-29 15:59

Contribution: 2021-01-29 16:00

Contribution: 2021-02-01 15:54

Contribution: 2021-02-01 15:55

Contribution: 2021-02-01 15:56

Contribution: 2021-02-02 15:54

Contribution: 2021-02-02 15:55

Contribution: 2021-02-02 15:56

Contribution: 2021-02-03 15:54

Contribution: 2021-02-03 15:55

Contribution: 2021-02-03 15:56

Contribution: 2021-02-05 15:54

Contribution: 2021-02-05 15:55

Contribution: 2021-02-05 15:56

Contribution: 2021-02-08 15:54

Contribution: 2021-02-08 15:55

Contribution: 2021-02-08 15:56

Contribution: 2021-02-08 15:57

Contribution: 2021-02-08 15:58

Contribution: 2021-02-08 15:59

Contribution: 2021-02-08 16:00

Contribution: 2021-02-09 15:54

Contribution: 2021-02-11 15:54

Contribution: 2021-02-11 15:55

Contribution: 2021-02-11 15:56

Contribution: 2021-02-11 15:57

Contribution: 2021-02-11 15:58

Contribution: 2021-02-11 15:59

Contribution: 2021-02-11 16:00

Contribution: 2021-02-12 15:54

Contribution: 2021-02-12 15:55

Contribution: 2021-02-15 15:54

Contribution: 2021-02-15 15:55

Contribution: 2021-02-15 15:56

Contribution: 2021-02-15 15:57

Contribution: 2021-02-15 15:58

Contribution: 2021-02-17 15:54

Contribution: 2021-02-17 15:55

Contribution: 2021-02-17 15:56

Contribution: 2021-02-17 15:57

Contribution: 2021-02-17 15:58

Contribution: 2021-02-17 15:59

Contribution: 2021-02-18 15:54

Contribution: 2021-02-18 15:55

Contribution: 2021-02-18 15:56

Contribution: 2021-02-18 15:57

Contribution: 2021-02-18 15:58

Contribution: 2021-02-18 15:59

Contribution: 2021-02-19 15:54

Contribution: 2021-02-19 15:55

Contribution: 2021-02-19 15:56

Contribution: 2021-02-19 15:57

Contribution: 2021-02-19 15:58

Contribution: 2021-02-19 15:59

Contribution: 2021-02-19 16:00

Contribution: 2021-02-22 15:54

Contribution: 2021-02-22 15:55

Contribution: 2021-02-22 15:56

Contribution: 2021-02-22 15:57

Contribution: 2021-02-22 15:58

Contribution: 2021-02-22 15:59

Contribution: 2021-02-22 16:00

Contribution: 2021-02-23 15:54

Contribution: 2021-02-23 15:55

Contribution: 2021-02-23 15:56

Contribution: 2021-02-25 15:54

Contribution: 2021-02-25 15:55

Contribution: 2021-02-25 15:56

Contribution: 2021-02-25 15:57

Contribution: 2021-02-25 15:58

Contribution: 2021-02-25 15:59

Contribution: 2021-02-26 15:54

Contribution: 2021-02-26 15:55

Contribution: 2021-02-26 15:56

Contribution: 2021-02-26 15:57

Contribution: 2021-02-26 15:58

Contribution: 2021-02-26 15:59

Contribution: 2021-03-01 15:54


Epanekkinisis/Assets/Sources/Scripts/Misc/Info.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class Info : MonoBehaviour {

	static public bool haschar;
	static public bool IsReg;

	static public int PlayerID;

	static public string Name;
	static public int Level;
	static public string Classi;
	static public string Sex;

	static public int Coins;
	static public int Xp;

	static public Dictionary<int, int> Levels = new Dictionary<int, int>();
	static public List<Quest> Quests = new List<Quest>();

	//----------------------------------------


	static public int GetNextLevel(int level){
		return Levels[level+1];
	}


	static public bool CheckLevel(int xp, int level){
		
		if(xp >= Levels[level + 1]){
			return true;
		}
		else{
			return false;
		}
		
	}

	
	static public void SetLevels(){
		
		Levels.Add(2, 100);
		Levels.Add(3, 400);
		Levels.Add(4, 800);
		Levels.Add(5, 1300);
		Levels.Add(6, 1900);
		Levels.Add(7, 2600);
		Levels.Add(8, 3400);
		Levels.Add(9, 4400);
		Levels.Add(10, 5400);
		Levels.Add(11, 6500);
		Levels.Add(12, 7700);
		Levels.Add(13, 9000);
		Levels.Add(14, 10400);
		Levels.Add(15, 11900);
		Levels.Add(16, 13500);
		Levels.Add(17, 15200);
		Levels.Add(18, 17000);
		Levels.Add(19, 18900);
		Levels.Add(20, 20900);
		Levels.Add(21, 23000);
		Levels.Add(22, 25200);
		Levels.Add(23, 27500);
		Levels.Add(24, 29900);
		Levels.Add(25, 32400);
		Levels.Add(26, 35000);
		Levels.Add(27, 35027);
		Levels.Add(28, 37827);
		Levels.Add(29, 40727);
		Levels.Add(30, 43727);
		Levels.Add(31, 46827);
		Levels.Add(32, 50027);
		Levels.Add(33, 53327);
		Levels.Add(34, 56727);
		Levels.Add(35, 60227);
		Levels.Add(36, 60263);
		Levels.Add(37, 60300);
		Levels.Add(38, 60338);
		Levels.Add(39, 60377);
		Levels.Add(40, 64377);
		Levels.Add(41, 105377);
		Levels.Add(42, 109577);
		Levels.Add(43, 113877);
		Levels.Add(44, 118277);
		Levels.Add(45, 122777);
		Levels.Add(46, 127377);
		Levels.Add(47, 132077);
		Levels.Add(48, 136877);
		Levels.Add(49, 141777);
		Levels.Add(50, 146777);
		Levels.Add(51, 151877);
		Levels.Add(52, 157077);
		Levels.Add(53, 162377);
		Levels.Add(54, 167777);
		Levels.Add(55, 173277);
		Levels.Add(56, 178877);
		Levels.Add(57, 184577);
		Levels.Add(58, 190377);
		Levels.Add(59, 196277);
		Levels.Add(60, 202277);
		Levels.Add(61, 208377);
		Levels.Add(62, 214577);
		Levels.Add(63, 220877);
		Levels.Add(64, 227277);
		Levels.Add(65, 233777);
		Levels.Add(66, 240377);
		Levels.Add(67, 247077);
		Levels.Add(68, 253877);
		Levels.Add(69, 260777);
		Levels.Add(70, 267777);

	}


	//To done ginete apo to network kai tsekarei gia to questtobedone sto npcquests.cs An den exei proigoumeno quest to questtobedoneid menei 0


	static public void SetQuests(){
	
		Quests.Add(new Quest());

		Quests[0].Finished = 1;

		// TO 0 KENO GIA N MI GAMITHEI KAI TO DONE TRUE GIA TA QUESTS POU DEN 8ELOUN QUESTTOBEDONEID
		// TO DONE KAI TO FINISHED APO TO MANAGER
		// To PlayerID iparxei mono sto db gia na ksexorizei tous pektes

		Quests.Add(new Quest());
				
		Quests[1].QuestID = 0;
		Quests[1].QuestNPC = "Saus";

		Quests[1].Name = "The beginning";
		Quests[1].Level = 1;

		Quests[1].Done = 0;
		Quests[1].Finished = 0;

		Quests[1].QuestToBeDoneID = 0;

		Quests[1].ADescription = "Quest description before player completes it.";
		Quests[1].BDescription = "Quest description after player completes it.";

		Quests[1].Xp = 100;
		Quests[1].Coins = 10;
		Quests[1].GivesItem = false;
		Quests[1].Reward = null;	//Allios me Resources.Load

	}


	static public void Start(){

		for (int i = 1; i <= Quests.Count-1; i++) {

			if(GameObject.Find(Quests[i].QuestNPC)){
				GameObject.Find(Quests[i].QuestNPC).GetComponent<NPC>().Quests.Add(Quests[i].QuestID);
			}

		}

	}


	static public int FindQuestFromID (int ID){

		int slot = 666;

		for(int i = 0; i <= Quests.Count-1; i++){

			if(Quests[i].QuestID == ID){
				slot = i;
			}

		}

		return slot;

	}

	
}
